packages/tokenbank_v1/contracts/lib/Authorizable.sol

```diff
@@ -0,0 +1,113 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../lib/Claimable.sol";
+
+
+/// @title Authorizable
+/// @author Brecht Devos - <brecht@loopring.org>
+/// @dev The Authorizable contract allows a contract to be used by other contracts
+///      by authorizing it by the contract owner.
+contract Authorizable is Claimable
+{
+    event AddressAuthorized(
+        address indexed addr
+    );
+
+    event AddressDeauthorized(
+        address indexed addr
+    );
+
+    // The list of all authorized addresses
+    address[] authorizedAddresses;
+
+    mapping (address => uint) private positionMap;
+
+    struct AuthorizedAddress {
+        uint    pos;
+        address addr;
+    }
+
+    modifier onlyAuthorized()
+    {
+        require(positionMap[msg.sender] > 0, "UNAUTHORIZED");
+        _;
+    }
+
+    function authorizeAddress(
+        address addr
+        )
+        external
+        onlyOwner
+    {
+        require(address(0) != addr, "ZERO_ADDRESS");
+        require(0 == positionMap[addr], "ALREADY_EXIST");
+        require(isContract(addr), "INVALID_ADDRESS");
+
+        authorizedAddresses.push(addr);
+        positionMap[addr] = authorizedAddresses.length;
+        emit AddressAuthorized(addr);
+    }
+
+    function deauthorizeAddress(
+        address addr
+        )
+        external
+        onlyOwner
+    {
+        require(address(0) != addr, "ZERO_ADDRESS");
+
+        uint pos = positionMap[addr];
+        require(pos != 0, "NOT_FOUND");
+
+        uint size = authorizedAddresses.length;
+        if (pos != size) {
+            address lastOne = authorizedAddresses[size - 1];
+            authorizedAddresses[pos - 1] = lastOne;
+            positionMap[lastOne] = pos;
+        }
+
+        authorizedAddresses.length -= 1;
```

Comment:
Don't delete authorizedAddresses[size-1]?
And authorizedAddresses not using now, seems positionMap is enough now, for future?

Reply:
authorizedAddresses is now made as public so it can be queried.